### PR TITLE
Add script to create Windows MSI installer

### PR DESCRIPTION
**Pull request by Siecje**
It packages the directory that is created by PyInstaller. I was not able to create the executable on Windows. (twisted failed to install for me on Windows 10 and Python 3.6).

Hopefully this is a good starting point and someone who has the PyInstaller executable can run this and create an MSI.

The MSI requires admin permission and installs into the Program Files directory.
There are lines commented out that will not require admin and will install into the users local AppData directory.

### CODE REVIEW

File: package/mu.wxs [hunk 1]
```diff
@@ -0,0 +1,48 @@
+<?xml version="1.0"?>
+
+<?define ProductName = "Mu Editor"?>
+<?define ProductVersion = "0.0.1"?>
+<!-- NEVER CHANGE THE upgrade_code BELOW!!!! -->
+<?define ProductUpgradeCode = "4981284B-DCB8-4257-91CA-D79EDC94A9E8"?>
+<?define CompanyName = "Trylan Erruh"?>
```
Comment (ZanderBrown) [inline package/mu.wxs]: Shouldn't this be mu-editor, ntoll or Micro:bit Educational Foundation?

File: package/mu.wxs [hunk 2]
```diff
@@ -0,0 +1,48 @@
+<?xml version="1.0"?>
+
+<?define ProductName = "Mu Editor"?>
+<?define ProductVersion = "0.0.1"?>
```
Comment (ZanderBrown) [inline package/mu.wxs]: I believe 1.0 beta would be more appropriate

File: package/mu.wxs [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+<?define InstallDir = "Mu Editor"?>
+
+<Wix xmlns="http://schemas.microsoft.com/wix/2006/wi">
+  <Product Id="*" UpgradeCode="$(var.ProductUpgradeCode)"
+           Name="$(var.ProductName)" Version="$(var.ProductVersion)" Manufacturer="$(var.CompanyName)" Language="1033">
+    <Package InstallerVersion="200" Compressed="yes" Comments="Windows Installer Package"/><!-- InstallScope="perUser" InstallPrivileges="limited"/> -->
+    <Media Id="1" Cabinet="product.cab" EmbedCab="yes"/>
+    <!-- <Icon Id="ProductIcon" SourceFile="example.ico"/> -->
+    <Property Id="ARPPRODUCTICON" Value="ProductIcon"/>
+    <Property Id="ARPHELPLINK" Value="https://codewith.mu/"/>
```
Comment (ZanderBrown) [inline package/mu.wxs]: Perhaps `https://codewith.mu/help/$(var.ProductVersion)`?

File: package/mu.wxs [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    <Property Id="ARPURLINFOABOUT" Value="https://codewith.mu/"/>
+    <Property Id="ARPNOREPAIR" Value="1"/>
+    <Property Id="ARPNOMODIFY" Value="1"/>
+    <Upgrade Id="$(var.ProductUpgradeCode)">
+      <UpgradeVersion Minimum="$(var.ProductVersion)" OnlyDetect="yes" Property="NEWERVERSIONDETECTED"/>
+      <UpgradeVersion Minimum="0.0.0" Maximum="$(var.ProductVersion)" IncludeMinimum="yes" IncludeMaximum="no"
+                      Property="OLDERVERSIONBEINGUPGRADED"/>
+    </Upgrade>
+    <Condition Message="A newer version of this software is already installed.">NOT NEWERVERSIONDETECTED</Condition>
```
Comment (ZanderBrown) [inline package/mu.wxs]: "`A newer version of $(var.ProductName) is already installed.`"

**Comment by ntoll** [comment]
Hi @Siecje,

This is wonderful work. Thank you! I'll take a look during the course of this week to try to make it a part of our automated build pipeline. @ZanderBrown ack all your points and I'll change as required.

Happy new year!

N.

**Comment by ntoll** [comment]
Hi @Siecje, thanks for the contribution. It turns out that we've been able to create a Windows installable via the `pynsist` tool, ergo I'll close this PR.

**Comment by carlosperate** [comment]
@ntoll Does this mean that MSI files are outside of the current Mu scope?
As far as I understand some IT managers needed that specific installer format to install remotely into mutiple systems at once.

**Comment by ntoll** [comment]
No, not just MSI, rather and "installable" (which we have).

**Pull request closed without merging**